java/trace/src/test/scala/org/typelevel/otel4s/java/trace/TracerSuite.scala

```diff
@@ -707,25 +712,37 @@ class TracerSuite extends CatsEffectSuite {
     val spanId = "279fa73bc935cc05"
 
     val headers = Map(
-      "traceparent" -> s"00-$traceId-$spanId-01"
+      "traceparent" -> s"00-$traceId-$spanId-01",
+      "foo" -> "1",
+      "baz" -> "2"
     )
 
     TestControl.executeEmbed {
       for {
-        sdk <- makeSdk()
+        sdk <- makeSdk(additionalPropagators =
+          Seq(PassThroughPropagator.create("foo", "bar"))
+        )
         tracer <- sdk.provider.get("tracer")
-        pair <- tracer.span("local").surround {
+        tuple <- tracer.span("local").surround {
           tracer.joinOrRoot(headers) {
-            tracer.currentSpanContext.product(
-              tracer.span("inner").use(r => IO.pure(r.context))
-            )
+            (
+              tracer.currentSpanContext,
+              tracer.span("inner").use(r => IO.pure(r.context)),
+              tracer.propagate(Map.empty[String, String]),
+            ).tupled
           }
         }
-        (external, inner) = pair
+        (external, inner, passthroughHeaders) = tuple
```

Comment:
nitpick: the trace headers aren't actually passthrough, so could we name this `resultHeaders` or something similar?